.pre-commit-config.yaml

```diff
@@ -32,3 +32,11 @@ repos:
     hooks:
       - id: cython-lint
         args: [--no-pycodestyle, --max-line-length=88]
+
+  - repo: local
+    hooks:
+      - id: pyproject.toml
+        name: pyproject.toml
+        language: system
+        entry: python tools/generate_pyproject.toml.py
```

Comment:
Presumably, this script has to be modified to compare the files and to return non-zero if any changes were made?

Reply:
It seems to work OK to me. It tries to regenerate the file and if it changes it fails see:
https://github.com/scikit-image/scikit-image/actions/runs/4209453776/jobs/7306442269

It will also fail if rtoml isn't installed as you would expect.

Reply:
Let me know if I am missing something or if you were just double-checking.

Reply:
It's neat that pre-commit detects these changes! Good to go then.